### PR TITLE
Fixed possible double events subscription

### PR DESCRIPTION
**Pull request by Karnah**
## What does the pull request do?
Does not allows subscribe on event twice if property value does not changed.
This very dangerous situation in WPF (because it often raise event even value does not changed) but almost impossible in Avalonia. I found only one case - if ```RaisePropertyChanged(Property, PropertyValue, PropertyValue);``` called directly.
I can not come up with case in real application with this bug. But if it happens i think it will hard to find problem.

## What is the current behavior?
Some methods can subscribe on event twice.

## What is the updated/expected behavior with this PR?
Correct unsubscription

## How was the solution implemented (if it's not obvious)?
I used old pattern for WPF:
```
if ((bool) e.OldValue)
    Unsubscribe();
if ((bool) e.NewValue)
    Subscribe();
```

This PR based on #2906

### CODE REVIEW

File: src/Avalonia.Controls/LayoutTransformControl.cs [hunk 1]
```diff
@@ -159,9 +159,16 @@ private void OnUseRenderTransformPropertyChanged(AvaloniaPropertyChangedEventArg
             //       workaround.
 
             var target = e.Sender as LayoutTransformControl;
+            var wasUsedRenderTransform = (bool)e.OldValue;
             var shouldUseRenderTransform = (bool)e.NewValue;
             if (target != null)
             {
+                if (wasUsedRenderTransform)
```
Comment (jmacato) [inline src/Avalonia.Controls/LayoutTransformControl.cs]: should be `hasUsedRenderTransform` perhaps?

**Review by jmacato — CHANGES_REQUESTED** [review]

File: src/Avalonia.Controls/LayoutTransformControl.cs (hunk 1 above)
Comment (Karnah) [reply src/Avalonia.Controls/LayoutTransformControl.cs]: Yes, you version is better

File: src/Avalonia.Controls/Button.cs [hunk 2]
```diff
@@ -382,17 +384,19 @@ private static void IsDefaultChanged(AvaloniaPropertyChangedEventArgs e)
         private static void IsCancelChanged(AvaloniaPropertyChangedEventArgs e)
         {
             var button = e.Sender as Button;
+            var wasCancel = (bool)e.OldValue;
```
Comment (jmacato) [inline src/Avalonia.Controls/Button.cs]:
```suggestion
            var wasCancelled = (bool)e.OldValue;
```

File: src/Avalonia.Controls/Button.cs [hunk 3]
```diff
@@ -382,17 +384,19 @@ private static void IsDefaultChanged(AvaloniaPropertyChangedEventArgs e)
...
             var isCancel = (bool)e.NewValue;
 
             if (button?.VisualRoot is IInputElement inputRoot)
             {
-                if (isCancel)
+                if (wasCancel)
```
Comment (jmacato) [inline src/Avalonia.Controls/Button.cs]:
```suggestion
                if (wasCancelled)
```

**Review by jmacato — CHANGES_REQUESTED** [review]
Sorry for the grammar nits, didn't notice them when i first reviewed this.

File: src/Avalonia.Controls/Button.cs (hunk 2 above)
Comment (Karnah) [reply src/Avalonia.Controls/Button.cs]: No problem, this is а good comment. Fixed and cleared history

**Comment by jmacato** [comment]
I can't decide if we should include this to 0.9 or not; @kekekeks & @danwalmsley what do you think about this PR?

**Comment by grokys** [comment]
I'm going to close this PR: calling a changed handler with an unchanged value is just something that you shouldn't do and should be considered a bug, so I don't think we need to go out of our way to handle this.

**Pull request closed without merging**